cmd/local-admin-client.go

```diff
@@ -80,3 +84,36 @@ func (lc localAdminClient) GetConfig() ([]byte, error) {
 
 	return json.Marshal(globalServerConfig)
 }
+
+// StartProfiling - starts profiling on the local server.
+func (lc localAdminClient) StartProfiling(profiler string) error {
+	if globalProfiler != nil {
+		globalProfiler.Stop()
+	}
+	globalProfiler = startProfiler(profiler, "")
+	return nil
+}
+
+// DownloadProfilingData - stops and returns profiling data of the local server.
+func (lc localAdminClient) DownloadProfilingData() ([]byte, error) {
+	if globalProfiler == nil {
+		return nil, errors.New("profiler not enabled")
+	}
+
+	profilerPath := globalProfiler.Path()
+
+	// Stop the profiler
+	globalProfiler.Stop()
+
+	profilerFile, err := os.Open(profilerPath)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := ioutil.ReadAll(profilerFile)
```

Comment:
I am not sure about this part - is there any bound on the file size? The reason to add profiling into admin apis is to profile in production - we should make sure that if profiling is left accidentally on for a long time, and collects a huge amount of profiling data, the step of reading all the data does not crash the server. In this line, if the profiling data is large, it may get OOM killed. At least it is worth checking if this is likely.